packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -1340,22 +1334,24 @@ export class AsgCapacityProvider extends CoreConstruct {
 class MaybeCreateCapacityProviderAssociations implements IAspect {
   private scope: CoreConstruct;
   private id: string;
-  private capacityProviders: AsgCapacityProvider[]
+  private capacityProviders: string[]
+  private resource?: CfnClusterCapacityProviderAssociations
 
-  constructor(scope: CoreConstruct, id: string, capacityProviders: AsgCapacityProvider[]) {
+  constructor(scope: CoreConstruct, id: string, capacityProviders: string[] ) {
     this.scope = scope;
     this.id = id;
     this.capacityProviders = capacityProviders;
   }
+
   public visit(node: IConstruct): void {
     if (node instanceof Cluster) {
```

Comment:
By the way this is instantiated, you *know* that `this.scope` is the `Cluster` you're looking for. You could do something like `if (node === this.scope)`.

Reply:
true. for whatever reason though, changing the condition to your suggestion causes a different complaint (clusterName not being on node, in line 1350).  will keep it as is since it works for now.